### PR TITLE
Add ARM MTE Support - May be broken

### PR DESCRIPTION
**Pull request by struct**
This is the start of MTE support in IsoAlloc. Most of the code is adapted from Scudo (also released under an Apache v2 license with minor modifications). It is UNTESTED at the moment and should not be merged. If you have a Pixel 8 and want to do some testing of this PR please get in touch with me.

### CODE REVIEW

**Comment by jvoisin** [comment]
cc @muhomorr @strcat

File: src/iso_alloc.c [hunk 1]
```diff
@@ -67,6 +67,16 @@ INTERNAL_HIDDEN void iso_alloc_initialize_global_root(void) {
      * result in a soft page fault */
     MLOCK(&_root, sizeof(iso_alloc_root));
 
+#if ARM_MTE
+    if(iso_is_mte_supported() == false) {
```
Comment (fmayer) [inline src/iso_alloc.c:71]: Random driveby: I would allow code that uses an allocator built with MTE support to also run on devices that don't support it. That simplifies deployment  a lot.

Comment (struct) [reply src/iso_alloc.c:71]: Random drive by appreciated!

File: include/iso_alloc_ds.h [hunk 2]
```diff
@@ -152,6 +152,9 @@ typedef struct {
     int32_t big_zone_free_count;
     int32_t big_zone_used_count;
     uint16_t zones_used;
+#if ARM_MTE
+    bool arm_mte_enabled;
```
Comment (jvoisin) [inline include/iso_alloc_ds.h:156]: This should go in a read-only zone I think, otherwise, MTE is disableable by an attacker with an arbitrary write primitive.

Comment (struct) [reply include/iso_alloc_ds.h:156]: A read-only section for settings like this is easy to implement but often requires dereferencing both `_root` and `ro_config_page` which can flush the cache and really hurt performance in any hot paths.

**Comment by struct** [comment]
This PR looks OK to merge behind the `ARM_MTE` build define which is disabled by default. Its mostly complete and tested using qemu. However I have little faith in the qemu implementation as I ran into several bugs involving memset() of `PROT_MTE` pages that don't appear to happen with a simple `while(){write()}` loop. Once I get access to a real device with MTE support I will retest this functionality.

TLDR: YMMV, please let me know if you can test this on real hardware!

**Comment by struct** [comment]
If you want to test in an Ubuntu Docker container on an M1/M2 based Macbook just enabled the `ARM_MTE` build defines in `Makefile` (make sure to set `DISABLE_CANARY=1` as well).

```
apt-get install clang-12 make qemu qemu-system-arm qemu-user-static
make mte_test
```

The `make` command above will compile test programs and just runs the following command:
```
qemu-aarch64-static -cpu max build/tests
```

If you add `-strace` to the qemu command you should be able to see it killed with the proper signal and `si_code` for `SEGV_MTESERR`.

```
--- SIGSEGV {si_signo=SIGSEGV, si_code=9, si_addr=0x0e00005501a15400} ---
qemu: uncaught target signal 11 (Segmentation fault) - core dumped
```
